src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -72,10 +73,10 @@
      * 3. Returning "true" will add the provided task to the list of updated tasks, if not there
      *    already.
      *
-     * @param other The other task
+     * @param that The other task
      * @return True if it can be, false if not
      */
-    boolean canRunConcurrentWith(AITask<O> other);
+    boolean canRunConcurrentWith(AITask<?> that);
```

Comment:
> If its changed to the ? then you put the pressure on the dev to ensure that the task you pass in is relevant to the task at hand.

Well, what are relevant tasks? IMO only those how could be executed concurrently which this method will determine.

> At least with what is there, you will be confident that the task being passed in will have an owner that is of the same type of your task's owner

I hope you will never check for a different entities' tasks with this method. IMO that method should only be called with matching owners. Using `<? super O>` is just garbage in the current setup because you cannot use the results any better than just `<?>` and you still have some of the compile errors from `<O>`. Using `<O>` has the potential future hazard that if you call a method that returns `XXXX<O>` for the other task you might get a `ClassCastException` since the real return type is only `XXXX<Agent>`. But more importantly. I don't see a usecase for being that specific since the only thing you can do with the generic type right now is the Owner, but since the owners match anyway you should not access the other one. On the other hand I must admit that there is currently no necessity to use `<?>` besides the compile issue. But the methods aren't supposed to be called that way anyway.

However there are a lot/some more generics issues in the AI API. (See also `Goal<O>.addTask(AITask<O> task)` that currently needs raw/unchecked casts, which is the entrance gate of evil)

We could solve this if we replace all methods parameters and results in `Goal` with <? super O> though => use `<?>` here
or replace all `___AITask<Creature>` with `___AITask<O extends Creature>` => Then we can use `<O>` here.
For all other solutions we have to do raw/unchecked casts in different places.